src/main/java/com/vmware/vipclient/i18n/messages/service/ComponentService.java

```diff
@@ -63,7 +63,8 @@ private void refreshCacheItem(final MessageCacheItem cacheItem, Iterator<DataSou
 		dto.setLocale(localeOrig);
 
 		// If timestamp is 0, it means that cacheItem not yet in cache. So try the next data source.
-		if (timestamp == 0) {
+		// If locale is "source", combine remote and local source messages.
+		if (timestamp == 0 || dto.getLocale().equals(ConstantsKeys.SOURCE)) {
```

Comment:
> Where to combine the locale and remote source messages?
This code already combines the remote and local source messages in the MessageCacheItem that was passed as a parameter.

Reply:
> Where to combine the locale and remote source messages?

Even before this change, the code already combines the remote and local source messages in the MessageCacheItem. Anything fetched from any datasource is added to cacheItem.cachedData using putAll. We do not delete anything from the cachedData map. For example, cacheItem was previously from remote, then it expires, then fetch from remote fails so it fetches from local. The result is a combination of remote and local in the cacheItem.

The only change here is not to wait for dataSource failure. Always combine source messages because source messages are more frequently updated (both in remote and local).

Reply:
Now we are implementing new feature of deleting a string from a bundle. Please take this into account in this PR.

Reply:
What do you mean by deleting a string from a bundle? Remote or offline? Either way, this sould not be done using the Java client library, just like source collection should not be done through the Java client library.